src/sentry/api/endpoints/organization_group_index_stats.py

```diff
@@ -0,0 +1,127 @@
+
+        The ``groupStatsPeriod`` parameter can be used to select the timeline
+        stats which should be present. Possible values are: '' (disable),
+        '24h', '14d'
+
+        The ``statsPeriod`` parameter can be used to select a date window starting
+        from now. Ex. ``14d``.
+
+        The ``start`` and ``end`` parameters can be used to select an absolute
+        date period to fetch issues from.
+
+        :qparam string statsPeriod: an optional stat period (can be one of
+                                    ``"24h"``, ``"14d"``, and ``""``).
+        :qparam string groupStatsPeriod: an optional stat period (can be one of
+                                    ``"24h"``, ``"14d"``, and ``""``).
+        :qparam string start:       Beginning date. You must also provide ``end``.
+        :qparam string end:         End date. You must also provide ``start``.
+        """
+
+        stats_period = request.GET.get("groupStatsPeriod")
+        try:
+            start, end = get_date_range_from_params(request.GET)
+        except InvalidParams as e:
+            raise ParseError(detail=six.text_type(e))
+
+        expand = request.GET.getlist("expand", [])
+        collapse = request.GET.getlist("collapse", ["base"])
+        has_inbox = features.has("organizations:inbox", organization, actor=request.user)
+        projects = self.get_projects(request, organization)
+        project_ids = [p.id for p in projects]
+
+        try:
+            group_ids = set(map(int, request.GET.getlist("groups")))
+        except ValueError:
+            return Response({"detail": "Group ids must be integers"}, status=400)
+
+        if not group_ids:
+            return Response(
+                {"detail": "You should include `groups` with your request. (i.e. groups=1,2,3)"},
+                status=400,
+            )
+        else:
+            groups = list(Group.objects.filter(id__in=group_ids, project_id__in=project_ids))
+            if not groups:
+                return Response({"detail": "No matching groups found"}, status=400)
+            elif len(groups) > 25:
+                return Response({"detail": "Too many groups requested."}, status=400)
+            elif any(g for g in groups if not request.access.has_project_access(g.project)):
+                raise PermissionDenied
+
+        if stats_period not in (None, "", "24h", "14d", "auto"):
+            return Response({"detail": ERR_INVALID_STATS_PERIOD}, status=400)
+        elif stats_period is None:
+            # default
+            stats_period = "24h"
+        elif stats_period == "":
+            # disable stats
+            stats_period = None
+
+        if stats_period == "auto":
+            stats_period_start = start
+            stats_period_end = end
+        else:
+            stats_period_start = None
+            stats_period_end = None
```

Comment:
Could this logic be shared with any of the other group endpoints? Seems like there'd be more copies of it about.

Reply:
Yeah, will definitely try to reuse this logic a bit more. Straight copy pasta'd this from the other endpoint.